### DISCUSSION TITLE
charge & discharge strategies, how to implement them?

### DISCUSSION THREAD

**Discussion by rrozema** [hif2k1/battery_sim#88] [Q&A]
I may have missed something in the installation/configuration instructions, but unless I explicitly press the "force charge" button, my simulated batteries just sit there doing nothing, remaining at the Empty level.

Do I need to configure or even script somehow that the battery starts charging automatically when my home delivers energy back into the grid (= returned solar energy)? And if so, how do I f.e. prevent that the battery charges at full speed, potentially charging off grid power, when less excess solar power is available than the battery's maximum charge speed?

**Comment by rrozema** [comment]
I think my initial problem could be the same as issue #83: I also have 2 tariffs set up. And I think in the past I have seen my simulated battery charging only on days that the sun was very bright very early, i.e. when tariff 1 was still active. Now that the days are less sunny, nothing happens at all any more, the batteries just sit there at Empty and never charge, even when I return loads of energy into the grid, the batteries don't charge.

**Comment by laplug** [comment]
I have the same problem. Did you find a solution?

**Comment by hif2k1** [comment]
Sorry, I didn't see this discussion till now. In the next release I am adding support for unlimited import/export tariffs and meters. I hope this will resolve your problem. If not then it must be something in the config, did you use the GUI setup or the yaml?

**Comment by rrozema** [comment]
I used the GUI to set up the plugin. But it's not so that I think the plugin doesn't work correctly. It is that I don't know how to automate it: I'm looking for an example or even an instruction on how to set up an automation that alters/amends the default plug-ins behavior of charging & discharging based on availability only.

My original problem -not charging the battery- was caused by me having 2 pairs of import and export sensors, the pair that reports power usage during the day time being the 2nd pair added which was ignored due to a bug that -I think- was fixed since then: After I created a pair of template sensors that sums the both tarif's usages and productions and used those as a single import & export pair, the battery started charging and discharging automatically.

**Comment by hif2k1** [comment]
The latest release supports unlimited imports and exports so I hope this resolves the issue without the need for creating extra sensors.
